include/RAJA/pattern/kernel/For.hpp

```diff
@@ -101,7 +101,9 @@ struct StatementExecutor<
     auto len = segment_length<ArgumentId>(data);
     using len_t = decltype(len);
 
-    forall_impl(ExecPolicy{}, TypedRangeSegment<len_t>(0, len), for_wrapper);
+    auto r = resources::get_resource<ExecPolicy>::type::get_default();
+
+    forall_impl(r, ExecPolicy{}, TypedRangeSegment<len_t>(0, len), for_wrapper);
```

Comment:
Not for this PR, but we might want to use wrap::forall instead of forall_impl in implementations here and there to get force inline.

Reply:
That makes sense, perhaps that can be fixed when resource support is added to kernel.